Misc Quality of Life and Bug fixes

Areloch opened the pull request:
Added handling for if preview images on image assets fails to generate, will fallback to using the full image
Added handling for double clicking or drag-n-dropping terrain assets to create them
Improved handling of field labels in variable inspector by making the stringtable be case sensitive.
Added editor settings for handling of asset double click behavior. Can now select between Edit Asset and Spawn Asset. Support is asset type dependent.
Added editor setting for auto-importing loose files when navigating to a folder. If on and the user has flagged to also enable auto-import generally, will auto import all unaffiliated loose files in as assets.
Added editor setting for default module to use when creating new assets. Updated various tooling logic so when creating a new material, if this and the 'Always Prompt Module Target' setting is off, it will fill in the target module and target asset path info based on the default module.
Fixed issue with editors that use managedData scripts where if the path didn't exist, the script file wouldn't be made.
Fixed display issue in terrain editor where if you clear the detail map, the normal/macro/orm maps would disable, but not also clear.
Fixed handling of cleared maps in terrain editor so it no longer fills empty maps in with the no image image.
Fixed handling of creating new material where it would fill in the diffuse with a no texture image as the default.
Fixed issue where canceling out of creating a module would still prompt to create the common default folders.
Fixed issue where the Select Module window couldn't be closed via the cancel or x buttons.
Based on feedback, reduced default size of the Text Pad window.
Fixed issue where the Drop At menu list wouldn't correctly display which item was marked after it was changed.
Fixed spawning shape asset handling so it uses whatever the editor's Drop At setting is.
Improved themeing of regular bitmap buttons in the editor.
Based on feedback, flipped layout of Target Module and Target Path in the Create New Asset window.
Improved handling of setting up the Target Path for when creating new assets. If a path is not set, and the user has a Default Module, it will default the path to that module.

Azaezel committed:
kill off glowchan leftovers

Azaezel committed:
nother stray

Azaezel committed:
be clear where we're referencing gbuffer render targets

Areloch committed:
Misc Quality of Life and Bug fixes
Added handling for if preview images on image assets fails to generate, will fallback to using the full image
Added handling for double clicking or drag-n-dropping terrain assets to create them
Improved handling of field labels in variable inspector by making the stringtable be case sensitive.
Added editor settings for handling of asset double click behavior. Can now select between Edit Asset and Spawn Asset. Support is asset type dependent.
Added editor setting for auto-importing loose files when navigating to a folder. If on and the user has flagged to also enable auto-import generally, will auto import all unaffiliated loose files in as assets.
Added editor setting for default module to use when creating new assets. Updated various tooling logic so when creating a new material, if this and the 'Always Prompt Module Target' setting is off, it will fill in the target module and target asset path info based on the default module.
Fixed issue with editors that use managedData scripts where if the path didn't exist, the script file wouldn't be made.
Fixed display issue in terrain editor where if you clear the detail map, the normal/macro/orm maps would disable, but not also clear.
Fixed handling of cleared maps in terrain editor so it no longer fills empty maps in with the no image image.
Fixed handling of creating new material where it would fill in the diffuse with a no texture image as the default.
Fixed issue where canceling out of creating a module would still prompt to create the common default folders.
Fixed issue where the Select Module window couldn't be closed via the cancel or x buttons.
Based on feedback, reduced default size of the Text Pad window.
Fixed issue where the Drop At menu list wouldn't correctly display which item was marked after it was changed.
Fixed spawning shape asset handling so it uses whatever the editor's Drop At setting is.
Improved themeing of regular bitmap buttons in the editor.
Based on feedback, flipped layout of Target Module and Target Path in the Create New Asset window.
Improved handling of setting up the Target Path for when creating new assets. If a path is not set, and the user has a Default Module, it will default the path to that module.

Azaezel committed:
DiffuseRenderPassManager.addManager "cleanups"
adds bintypes where they're lacking, since that's an active object-type filter to avoid accidently rendering things twice. (particleBin being like glowmanager, a seperate flow, that acts up if filtered in that manner)
basiconly likewise is intended to avoid multiple-pass insertions, despite the varname.

JeffProgrammer commented on Templates/BaseGame/game/tools/worldEditor/scripts/menuHandlers.ed.tscript:
```diff
@@ -1059,7 +1059,12 @@ function EditorDropTypeMenu::onSelectItem(%this, %id, %text)
    EWorldEditor.dropType = getField(%this.item[%id], 2);
    
-   %this.checkRadioItem(0, (%this.getItemCount() - 1), %id);
+   for(%i=0; %i < %this.getItemCount() + 1; %i++)
```
item count + 1? Does this go over the array?

Azaezel committed:
Merge pull request #549 from Azaezel/alpha40/shadergenCleanups

be clear where we're referencing gbuffer render targets

Azaezel committed:
Merge pull request #548 from Azaezel/alpha40/glowchanCleanups

kill off glowchan leftovers

Azaezel committed:
ensure new (foo) lines being converted are valid according to findObjectClass

Azaezel committed:
sanity check nodelist presence

JeffProgrammer committed:
Merge pull request #553 from Azaezel/alpha40/nodelistnonull

sanity check nodelist presence

Azaezel committed:
account for the possiblity of _set##name(StringTableEntry _in entries somehow getting punted nulls

Azaezel committed:
ironically, overly paranoid checks were tripping warnings

JeffProgrammer committed:
Update thread ids for 64bit support.

lukaspj commented on Templates/BaseGame/game/tools/assetBrowser/scripts/addModuleWindow.tscript:
```diff
@@ -32,7 +32,7 @@ function AssetBrowser_addModuleWindow::onGainFirstResponder(%this)
 {
    Canvas.popDialog(AssetBrowser_addModule);  
-   eval(AssetBrowser_addModuleWindow.callbackFunction);
+   //eval(AssetBrowser_addModuleWindow.callbackFunction);
```
Might as well just delete the line

lukaspj commented on Templates/BaseGame/game/tools/assetBrowser/scripts/assetTypes/image.tscript:
```diff
@@ -196,22 +196,35 @@ function AssetBrowser::buildImageAssetPreview(%this, %assetDef, %previewData)
       %success = saveScaledImage(%assetDef.getImagePath(), %previewFilePath);
       
+      if(%success)
+      {
```
Indentation below this line looks off

lukaspj commented on Templates/BaseGame/game/tools/assetBrowser/scripts/assetTypes/shape.tscript:
```diff
@@ -318,16 +318,23 @@ function AssetBrowser::buildShapeAssetPreview(%this, %assetDef, %previewData)
+      if(EditorSettings.value("Assets/Browser/doubleClickAction", "Edit Asset") $= "Edit Asset")
       %previewData.doubleClickCommand = "AssetBrowser.editAsset( "@%assetDef@" );";
+      else
+         %previewData.doubleClickCommand = "AssetBrowser.onShapeAssetEditorDropped( "@%assetDef@" );";
```
Generally I think if you don't have braces it should be a single line. I'd prefer braces here

lukaspj commented on Templates/BaseGame/game/tools/assetBrowser/scripts/assetTypes/shape.tscript:
```diff
@@ -340,7 +347,7 @@ function AssetBrowser::onShapeAssetEditorDropped(%this, %assetDef, %position)
    {
       %pos = "0 0 0";  
-   }
+   }*/
```
Might as well delete the code

lukaspj commented on Templates/BaseGame/game/tools/gui/profiles.ed.tscript:
```diff
@@ -353,13 +353,18 @@ new GuiControlProfile( ToolsGuiButtonProfile )
    justify = "center";
    canKeyFocus = false;
-	bitmapAsset = "ToolsModule:button_image";
+	//bitmapAsset = "ToolsModule:button_image";
```
Delete This line instead

Azaezel committed:
Merge pull request #552 from Azaezel/alpha40/classconvertsafety

ensure new (foo) lines being converted are valid according to findObj…

Azaezel committed:
Merge pull request #551 from Azaezel/alpha40/decalDemangle

DiffuseRenderPassManager.addManager "cleanups"

Azaezel committed:
Merge pull request #554 from Azaezel/alpha40/setNULLnono

account for the possiblity of _set##name(StringTableEntry _in entries somehow getting punted nulls

Azaezel committed:
mini cleanups for ab
seems some operating systems don't play nice with setting `extent =` outside of a define{init} block,
cleaned up %assetarray using $AssetBrowser::AssetArray
cleanup for terrain definition path display

Areloch committed:
Improves handling of rendering guiPopupCtrls where if the height extent is taller than the bitmap array height, it'll adjust the height to recenter the displayed bitmap elements.
Streamlined the toolbar for the gui and world editors to utilize a stack, making the behavior and manipulation of toolbar elements significantly more consistent.
Added Settings and Asset Browser buttons to both gui and world editor toolbars for easier access.
Moved all tool toolbars over to work with the stack system to make them more consistent and better formatting
Added saving of asset browser's last position and extent so it remembers it on load.
Added editor setting to close the asset browser after completing a drag-n-drop action.
Added keybind to editor keybind list, making space toggle the asset browser

Areloch committed:
Missed line adding ShapeEditorToolbar to main editor UI needlessly.

Azaezel committed:
Merge pull request #555 from JeffProgrammer/threadid

Update thread ids for 64bit support.

Azaezel committed:
Merge pull request #556 from Azaezel/alpha40/minorABcleanups

mini cleanups for ab

Azaezel committed:
Merge pull request #558 from Areloch/MiscFixes20210901

Misc Fixes and QOL improvements

Areloch committed:
Merge branch 'MiscQOLFixes20210828' of https://github.com/Areloch/Torque3D into Preview4_0

Areloch committed:
Requested cleanups
